tests/vm_test.go

```diff
@@ -1921,6 +1924,63 @@ status:
 			}, 2*time.Minute, 1*time.Second).Should(Succeed())
 		})
 	})
+
+	Context("[Serial] when node becomes unhealthy", Serial, func() {
+		const componentName = "virt-handler"
+
+		getHandlerNodePod := func(nodeName string) *k8sv1.PodList {
+			pods, err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.Background(),
+				metav1.ListOptions{
+					LabelSelector: fmt.Sprintf("kubevirt.io=%s", componentName),
+					FieldSelector: fmt.Sprintf("spec.nodeName=" + nodeName),
+				})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(pods.Items).To(HaveLen(1))
+
+			return pods
+		}
+
+		It("[Serial] the VMs running in that node should be respawned", func() {
+
+			By("Starting VM")
+			vm := startVM(virtClient, createVM(virtClient, libvmi.NewCirros()))
+			vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			nodeName := vmi.Status.NodeName
+			oldUID := vmi.UID
+
+			By("Blocking virt-handler from reconciling the VMI")
+			libpod.AddKubernetesApiBlackhole(getHandlerNodePod(nodeName), componentName)
+			Eventually(getHandlerNodePod(nodeName).Items[0], 120*time.Second, time.Second, HaveConditionFalse(k8sv1.PodReady))
+
+			DeferCleanup(func() {
+				libpod.DeleteKubernetesApiBlackhole(getHandlerNodePod(nodeName), componentName)
+				Eventually(getHandlerNodePod(nodeName).Items[0], 120*time.Second, time.Second, HaveConditionTrue(k8sv1.PodReady))
+			})
+
+			pod, err := libvmi.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Simulating loss of the virt-launcher")
+			err = virtClient.CoreV1().Pods(pod.Namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{
+				GracePeriodSeconds: ptr.To(int64(0)),
+			})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("The VM should not be Ready")
+			Eventually(ThisVM(vm), 60*time.Second, 1*time.Second).ShouldNot(beReady())
+
+			By("The controller should recreate the launcher pod and move the VMI to phase Scheduled")
+			Eventually(ThisVMI(vmi), 60*time.Second, 2*time.Second).Should(BeInPhase(v1.Scheduled))
```

Comment:
Did you try out lower values for the timeout? We want it to be fast. If this slows down significantly it could be considered a regression. Maybe add a comment about it here.